includes/gateways/stripe/controllers/classStrpEventsController.php

```diff
@@ -0,0 +1,377 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	die( 'You are not allowed to call this page directly.' );
+}
+
+class WPBDPStrpEventsController {
+
+	/**
+	 * @var string
+	 */
+	public static $events_to_skip_option_name = 'wpbdp_strp_events_to_skip';
+
+	private $event;
+	private $invoice;
+
+	/**
+	 * Tell Stripe Connect API that the request came through by flushing early before processing.
+	 * Flushing early allows the API to end the request earlier.
+	 *
+	 * @since x.x
+	 *
+	 * @return void
+	 */
+	private function flush_response() {
+		ob_start();
+
+		// Get the size of the output.
+		$size = ob_get_length();
+
+		// Disable compression (in case content length is compressed).
+		header( 'Content-Encoding: none' );
+
+		// Set the content length of the response.
+		header( 'Content-Length: ' . $size );
+
+		// Close the connection.
+		header( 'Connection: close' );
+
+		// Flush all output.
+		ob_end_flush();
+		@ob_flush();
+		flush();
+	}
+
+	/**
+	 * @return void
+	 */
+	public function process_connect_events() {
+		$this->flush_response();
+
+		$unprocessed_event_ids = WPBDPStrpConnectHelper::get_unprocessed_event_ids();
+		if ( $unprocessed_event_ids ) {
+			$this->process_event_ids( $unprocessed_event_ids );
+		}
+		wp_send_json_success();
+	}
+
+	/**
+	 * @since x.x
+	 *
+	 * @param array<string> $event_ids
+	 *
+	 * @return void
+	 */
+	private function process_event_ids( $event_ids ) {
+		foreach ( $event_ids as $event_id ) {
+			if ( $this->should_skip_event( $event_id ) ) {
+				continue;
+			}
+
+			set_transient( 'wpbdp_last_process_' . $event_id, time(), 60 );
+
+			$this->event = WPBDPStrpConnectHelper::get_event( $event_id );
+			if ( is_object( $this->event ) ) {
+				$this->handle_event();
+				$this->track_handled_event( $event_id );
```

Comment:
I think I prefer it this way. It reduces the complexity of each of the functions having the logic separated and allows us to test each thing in isolation (ie. I could write tests for `handle_event` and `track_handled_event` separately instead of having to test both things together).

Since `handle_event` is a larger function I think it would also be easier to miss that `track_handled_event` is called at the bottom of the `switch/case`.